service/src/test/java/bio/terra/workspace/service/resource/controlled/ControlledResourceMetadataManagerTest.java

```diff
@@ -0,0 +1,125 @@
+package bio.terra.workspace.service.resource.controlled;
+
+import static bio.terra.workspace.service.resource.model.WsmResourceState.NOT_EXISTS;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import bio.terra.common.exception.ForbiddenException;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.common.exception.InternalLogicException;
+import bio.terra.workspace.common.fixtures.ControlledResourceFixtures;
+import bio.terra.workspace.db.ApplicationDao;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.model.SamConstants;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer.ControlledAzureStorageContainerResource;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.stage.StageService;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+public class ControlledResourceMetadataManagerTest extends BaseUnitTest {
```

Comment:
There were no unit tests for this class-- it was pulled out of someplace else for reuse, and some connected or integration tests probably hit it incidentally. I didn't want to try to add to a connected test, so created a pure unit test for the purposes of testing this specific codepath.